### DISCUSSION TITLE
How to make use of complex Renovate Scheduling?

### DISCUSSION THREAD

**Discussion by kirselandise** [renovatebot/renovate#11098] [Request Help]
My team requires Renovate to do a specific schedule for dependency updates. We want multiple groups that update on alternating weekends. This is due to our large number of repositories, we can't just have them update at the same time. 

So we have `groupA` updating the first weekend of a month and alternates every other. Then `groupB` on the second weekend, alternating opposite of `groupA`. So one weekend is A, then B, then A, etc.

We're struggling to find the correct way to schedule this:

For groupA we currently have: `"on the first saturday of the month", "on the 3rd saturday of the month"`
though "`3rd`" doesn't appear to work.

Does anybody have specific guidelines or rules for the scheduling system. Or any idea how to get more complex scheduling system like this going

**Comment by rarkins** [comment]
I don't know if you have any free resources, but implementing https://github.com/renovatebot/renovate/issues/7612 would probably make your life much easier. Otherwise we are bound by the capabilities of the `later` library we depend on.

**Reply by kirselandise** [reply]
Thanks for replying
Do you have any idea where I could look into `later` syntax? If something like `"on the 2nd Saturday of every month"` would work with it, that would be amazing, I just need somewhere to look at the rules.

**Reply by viceice** [reply]
👉 https://docs.renovatebot.com/configuration-options/#schedule

Bottom of schedule config 👀

![image](https://user-images.githubusercontent.com/1798109/128308230-0cb34f22-c61f-418a-a021-21b77d984eb6.png)

https://breejs.github.io/later/parsers.html#text

**Reply by rarkins** [reply]
You might also find it useful to look at the unit tests: https://github.com/renovatebot/renovate/blob/main/lib/workers/branch/schedule.spec.ts

**Reply by kirselandise** [reply]
> 👉 https://docs.renovatebot.com/configuration-options/#schedule
> 
> Bottom of schedule config 👀
> 
> ![image](https://user-images.githubusercontent.com/1798109/128308230-0cb34f22-c61f-418a-a021-21b77d984eb6.png)
> 
> https://breejs.github.io/later/parsers.html#text

This is exactly the ruleset we already looked at, but it doesn't appear to work with Renovate

Here's three separate instances we've tried, and all report back errors:
```
WARN: Invalid schedule: Failed to parse "on the second saturday of the month"
WARN: Invalid schedule: Failed to parse "on the 2nd saturday of the month"
WARN: Invalid schedule: Failed to parse "on the 2nd saturday of every month"
```
I've already looked through that link before and came to these attempting to follow those guidelines, but they still don't work. 

So if anyone knows what issue I'm running into, and why, that would be very helpful
I could very easily just be misunderstanding the document, and that's the issue I would like to solve

**Reply by timnolte** [reply]
@kirselandise did you ever figure out a solution to your scheduling needs? I was looking to try to make sure that PRs were opened using the schedule `"before 6am on the first weekday of the month"`, but based on the examples you gave I'm thinking it's not going to work.

**Reply by timnolte** [reply]
Looking at the details(https://breejs.github.io/later/parsers.html#text) on how things like `first` are parsed and their meaning I think I understand why why we are trying to accomplish doesn't work. In my case both `first` and `weekday` represent an `on()` call. You can't have an `on().on()` call there needs to be a Time Period for each `on()` call.
